src/NuGet.Clients/NuGet.VisualStudio.Implementation/Extensibility/NuGetProjectService.cs

```diff
@@ -95,7 +96,19 @@ NuGetInstalledPackage ToNuGetInstalledPackage(PackageReference packageReference,
             IReadOnlyCollection<NuGetInstalledPackage> installedPackages;
 
             var cacheContext = new DependencyGraphCacheContext();
-            var (packageSpecs, messages) = await project.GetPackageSpecsAndAdditionalMessagesAsync(cacheContext);
+            IReadOnlyList<ProjectModel.PackageSpec> packageSpecs;
+            IReadOnlyList<ProjectModel.IAssetsLogMessage> messages;
+            try
+            {
+                (packageSpecs, messages) = await project.GetPackageSpecsAndAdditionalMessagesAsync(cacheContext);
+            }
+            catch (ProjectNotNominatedException)
```

Comment:
Looking at the below logic, when `ProjectNotNominatedException` exception is raised, we pass some additional information for `Message` property which is ignored/not logged in the exception handler. Does it help if we log the message somewhere?

```cs
 throw new ProjectNotNominatedException(
                    string.Format(Strings.ProjectNotLoaded_RestoreFailed, ProjectName));
```

Reply:
`CpsPackageReferenceProject` is used in many, many places. `NuGetProjectService.cs` might not use the exception message, but other code paths will.